core/src/main/java/hudson/node_monitors/AbstractAsyncNodeMonitorDescriptor.java

```diff
@@ -101,11 +111,46 @@ protected T monitor(Computer c) throws IOException, InterruptedException {
                 } catch (TimeoutException x) {
                     LOGGER.log(WARNING, "Failed to monitor " + c.getDisplayName() + " for " + getDisplayName(), x);
                 }
+            } else {
+                skipped.add(c);
             }
         }
 
-        return data;
+        return new ResultMap<>(data, skipped);
     }
 
     private static final Logger LOGGER = Logger.getLogger(AbstractAsyncNodeMonitorDescriptor.class.getName());
+
+    /**
+     * Crate returned by {@link AbstractAsyncNodeMonitorDescriptor#monitor()}, it extends Map as that is what the interface
```

Comment:
It just struck me that this is probably another typo. I think "Crate" means "Create", though the rest of the line is still a little confusing.

Reply:
Heh, my bad. It seems the "Crate design pattern" has made its way to no more than a single publication on design patterns (not even a good one) without wider adoption.

The class will serve as a "bag" to get several conceptual values through an interface where single value is expected/desired allowing for future evolution of the API without breaking the method API.

I will rephrase that, suggestions welcome.

Reply:
Devin and I wondered about something like that, but I couldn't find any reference to that sort of usage anywhere. I didn't look very hard so that must be why I missed that one instance. British English may be different from my American English, but I've never heard "crate" used in that sort of form.

An alternative word might be "container", as in "Container returned by ...". Or if you want to maintain the naming from that one design pattern, it might be clearer as "Crate (or container)". 

To be really picky, there should be different punctuation on that line. Something like, "Crate returned by {@link AbstractAsyncNodeMonitorDescriptor#monitor()}. It extends Map as that is what the interface ...". Or possibly, "Crate returned by {@link AbstractAsyncNodeMonitorDescriptor#monitor()}; it extends Map as that is what the interface ...". I'm not worried about this bit, though. The meaning is still plenty clear as it is, but since we were talking about this line, I thought I'd mention it.